lib/restforce/middleware/authentication/jwt_bearer.rb

```diff
@@ -0,0 +1,28 @@
+require 'jwt'
+
+module Restforce
+  # Authentication middleware used if oauth_token and refresh_token are set
+  class Middleware::Authentication::JWTBearer < Restforce::Middleware::Authentication
+    def params
+      { grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+        assertion: jwt_bearer_token }
+    end
+
+    def jwt_bearer_token
+      JWT.encode claim_set, private_key, 'RS256'
+    end
+
+    def claim_set
+      {
+          iss: @options[:client_id],
+          sub: @options[:username],
+          aud: @options[:host],
```

Comment:
hi @tagCincy 

I think you should pass @options[:instance_url] as aud parameter. In case of passing just a host 'aud' parameter doesn't get  protocol prefix and Salesforce is getting back with 'invalid audience' error. 

As stated on https://help.salesforce.com/articleView?id=remoteaccess_oauth_jwt_flow.htm&type=5

The audience (aud) must be https://login.salesforce.com, https://test.salesforce.com, or, if implementing for a community, https://acme.force.com/customers (where acme.force.com/customers is your community URL).